lib/src/mapbox_map.dart

```diff
@@ -754,6 +756,28 @@ class MapboxMap extends ChangeNotifier {
   @experimental
   Future<void> setSnapshotLegacyMode(bool enable) =>
       _mapInterface.setSnapshotLegacyMode(enable);
+
+  /// Set custom headers for all Mapbox HTTP requests
+  ///
+  /// [headers] is a map of header names to header values
+  ///
+  /// Throws a [PlatformException] if the native implementation is not available
+  /// or if the operation fails
+  ///
+  /// Example:
+  /// ```dart
+  /// MapboxMap.setCustomHeaders({
+  ///   "Authorization": "Bearer sk.eyJ1IjoiYWxleG9yZ2V0IiwiYSI6ImNtZ291Z291MjB1ZjIyc251Zm91Zm91ZjEifQ.1234567890",
```

Comment:
This is a secret [token](https://docs.mapbox.com/accounts/guides/tokens/). If someone else gets access to your tokens with secret scopes they may be able to make changes to your account.

Please [rotate](https://docs.mapbox.com/accounts/guides/tokens/#rotating-access-tokens) this token as soon as possible.